src/module.cpp

```diff
@@ -654,32 +632,66 @@ void module::print_cpp(std::ostream& os) const
 
 void module::annotate(std::ostream& os, std::function<void(instruction_ref)> a) const
 {
-    this->print([&](auto ins, const auto& names) {
-        print_instruction(os, ins, names);
+    this->print([&](auto ins, auto ins_names) {
+    	instruction::print(os, ins, ins_names);
         a(ins);
         os << std::endl;
     });
 }
 
+std::vector<module_ref> module::get_sub_modules() const
+{
+    std::vector<module_ref> vec_modules;
+    this->print([&](auto ins, auto) {
```

Comment:
Why are we calling `print` to get submodules? We should just iterate over the instructions.

Reply:
changed